tests/js/src/test/scala/io/circe/scalajs/ScalaJSSuite.scala

```diff
@@ -0,0 +1,38 @@
+package io.circe.scalajs
+
+import io.circe.tests.CirceSuite
+import io.circe.generic.auto._
+import scala.scalajs.js
+import scalajs.js.Dynamic.{literal => sjson}
+
+case class Sample(name: String)
+
+case class Sample2(name: js.UndefOr[String])
+
+class ScalaJSSuite extends CirceSuite {
+
+  test("should decode js.Object to Scala Class") {
```

Comment:
I'd prefer to use property-based testing via Scalacheck in these tests, but we could open a separate issue for that.